manager/src/main/java/com/redhat/service/bridge/manager/ProcessorServiceImpl.java

```diff
@@ -65,12 +73,26 @@ public Processor getProcessor(String processorId, String bridgeId, String custom
     @Transactional
     @Override
     public Processor createProcessor(String bridgeId, String customerId, ProcessorRequest processorRequest) {
-        Bridge bridge = getAvailableBridge(bridgeId, customerId);
+        /* We cannot deploy Processors to a Bridge that is not Available */
+        Bridge bridge = bridgesService.getAvailableBridge(bridgeId, customerId);
+
         if (processorDAO.findByBridgeIdAndName(bridgeId, processorRequest.getName()) != null) {
             throw new AlreadyExistingItemException("Processor with name '" + processorRequest.getName() + "' already exists for bridge with id '" + bridgeId + "' for customer '" + customerId + "'");
         }
 
-        ProcessorDefinition definition = new ProcessorDefinition(processorRequest.getFilters(), processorRequest.getTransformationTemplate(), processorRequest.getAction());
+        final Set<BaseFilter> requestFilters = processorRequest.getFilters();
+        final String requestTransformationTemplate = processorRequest.getTransformationTemplate();
+        final BaseAction requestAction = processorRequest.getAction();
+
+        Optional<BaseAction> optTransformedAction = actionProviders.stream().filter(a -> a.accept(requestAction.getType())).findFirst()
```

Comment:
@kostola For me, this is not the right separation of concerns. I think the `ProcessorServiceImpl` should simply ask the `ActionProviderFactory` to give it the resolved Action. It shouldn't care how the Action is resolved.

So I see a workflow here, something like:

```
BaseAction resolvedAction = actionProviderFactory.resolveAction(processorRequest.getAction());

ProcessorDefinition d = new ProcessorDefinition();
d.setRequestedAction(processorRequest.getAction());
d.setResolvedAction(resolvedAction);
```

Reply:
@robpblake I pushed a change that tries to replicate the implementation you described...please take a look :wink: